src/Agent.Sdk/Knob/AgentKnobs.cs

```diff
@@ -63,6 +63,13 @@ public class AgentKnobs
             new EnvironmentKnobSource("VSTS_DISABLEFETCHBYCOMMIT"),
             new BuiltInDefaultKnobSource("false"));
 
+        public static readonly Knob DisableFetchPruneTags = new Knob(
+            nameof(DisableFetchPruneTags),
+            "If true, disable --prune-tags in the initial fetch.",
```

Comment:
Is this only for initial fetch? Will it be the same for several checkout task in pipeline?

Reply:
Other checkout shall be the same, probaby need to reword it